src/loader.c

```diff
@@ -32,6 +32,60 @@ static volatile const uint32_t __attribute__((used)) wolfboot_version = WOLFBOOT
 extern void (** const IV_RAM)(void);
 #endif
 
+#if defined(WOLFBOOT_PARTION_INFO) && defined(PRINTF_ENABLED)
+
+#include <printf.h>
+
+static void printPart(uint8_t *part)
+{
+#ifdef WOLFBOOT_PARTION_VERBOS
+    uint32_t *v;
+    int i;
+#endif
+    uint8_t  *magic;
+    uint8_t  state;
+    uint32_t ver;
+
+    magic = part;
+    wolfBoot_printf("Magic:    %c%c%c%c\n", magic[0], magic[1], magic[2], magic[3]);
+    ver = wolfBoot_get_blob_version(part);
+    wolfBoot_printf("Version:  %02x\n", ver);
+    state = *(part + WOLFBOOT_PARTITION_SIZE - sizeof(uint32_t) - 1);
+    wolfBoot_printf("Status:   %02x\n", state);
+    magic = part + WOLFBOOT_PARTITION_SIZE - sizeof(uint32_t);
+    wolfBoot_printf("Tail Mgc: %c%c%c%c\n", magic[0], magic[1], magic[2], magic[3]);
+
+#ifdef WOLFBOOT_PARTIION_VERBOS
+    v = (uint32_t *)part;
+    for(i = 0; i < 0x100/4; i++) {
+        if(i % 4 == 0)
+            wolfBoot_printf("\n%08x: ", (uint32_t)v+i*4);
```

Comment:
location is not guaranteed to be fixed. You should use TLV parser to retrieve versions (e.g. `wolfBoot_current_firmware_version` etc.)